tests/std/tests/P2374R4_views_cartesian_product_recommended_practices/test.compile.pass.cpp

```diff
@@ -0,0 +1,473 @@
+    static_assert(_Compile_time_max_size<V2> == 500'000'000);
+    static_assert(_Compile_time_max_size<V3> == 500'000'000);
+
+    static_assert(
+        sized_range<V1> == (sized_range<V2> && sized_range<V3>) ); // If V1 is sized, then all of them should be
+    if constexpr (sized_range<V1>) {
+        static_assert(sizeof(cpv_size_t<V1>) <= sizeof(size_t));
+        static_assert(sizeof(cpv_size_t<V1, V1>) <= sizeof(size_t));
+        static_assert(sizeof(cpv_size_t<V2>) <= sizeof(size_t));
+        static_assert(sizeof(cpv_size_t<V2, V2>) <= sizeof(size_t));
+        static_assert(sizeof(cpv_size_t<V3>) <= sizeof(size_t));
+        static_assert(sizeof(cpv_size_t<V3, V3, V3>) > sizeof(size_t));
+    }
+
+    static_assert(sizeof(cpv_difference_t<V1>) <= sizeof(ptrdiff_t));
+    static_assert(sizeof(cpv_difference_t<V1, V1>) <= sizeof(ptrdiff_t));
+    static_assert(sizeof(cpv_difference_t<V2>) <= sizeof(ptrdiff_t));
+    static_assert(sizeof(cpv_difference_t<V2, V2>) <= sizeof(ptrdiff_t));
+    static_assert(sizeof(cpv_difference_t<V3>) <= sizeof(ptrdiff_t));
+    static_assert(sizeof(cpv_difference_t<V3, V3, V3>) > sizeof(ptrdiff_t));
+
+    static_assert(
+        to_underlying(CCA) == (ranges::range<const V1> && ranges::range<const V2> && ranges::range<const V3>) );
+    if constexpr (to_underlying(CCA)) {
+        static_assert(_Compile_time_max_size<const V1> == 100);
+        static_assert(_Compile_time_max_size<const V2> == 500'000'000);
+        static_assert(_Compile_time_max_size<const V3> == 500'000'000);
+
+        if constexpr (sized_range<const V1>) {
+            static_assert(sizeof(cpv_const_size_t<V1>) <= sizeof(size_t));
+            static_assert(sizeof(cpv_const_size_t<V1, V1>) <= sizeof(size_t));
+        }
+
+        if constexpr (sized_range<const V2>) {
+            static_assert(sizeof(cpv_const_size_t<V2>) <= sizeof(size_t));
+            static_assert(sizeof(cpv_const_size_t<V2, V2, V2>) > sizeof(size_t));
+        }
+
+        if constexpr (sized_range<const V3>) {
+            static_assert(sizeof(cpv_const_size_t<V3>) <= sizeof(size_t));
+            static_assert(sizeof(cpv_const_size_t<V3, V3, V3>) > sizeof(size_t));
+        }
+
+        static_assert(sizeof(cpv_const_difference_t<V1>) <= sizeof(ptrdiff_t));
+        static_assert(sizeof(cpv_const_difference_t<V1, V1>) <= sizeof(ptrdiff_t));
+        static_assert(sizeof(cpv_const_difference_t<V2>) <= sizeof(ptrdiff_t));
+        static_assert(sizeof(cpv_const_difference_t<V2, V2, V2>) > sizeof(ptrdiff_t));
+        static_assert(sizeof(cpv_const_difference_t<V3>) <= sizeof(ptrdiff_t));
+        static_assert(sizeof(cpv_const_difference_t<V3, V3, V3>) > sizeof(ptrdiff_t));
+    }
+}
+
+constexpr void check_join_view() {
+    using V1 = ranges::join_view<all_t<array<array<int, 10>, 20>>>;
+    using V2 = ranges::join_view<all_t<array<span<int, 5'000'000>, 100>>>;
+    using V3 = ranges::join_view<all_t<span<span<int, 10'000'000'000>, 5'000'000'000>>>;

```

Comment:
test.compile.pass.cpp(237): warning C4309: 'specialization': truncation of constant value.

32 bit `std::size_t` can't store `10'000'000'000`

Reply:
Yep, I'm already working on fixing this.